### ISSUE TITLE
Please fix breakage caused by removal of `extra_env`

### ISSUE THREAD

**Issue by ntoll** [jupyter/jupyter_client#365]
It used to be possible to do something like the following:

```
self.repl_kernel_manager = QtKernelManager(extra_env=self.envars)
self.repl_kernel_manager.start_kernel()
```

...and the dict `self.envars` passed in as `extra_env` would add environment variables to `os.environ` before the kernel started.

(See: https://github.com/jupyter/jupyter_client/blob/master/jupyter_client/manager.py#L244)

This appears to have been removed with no mention in the changes. :-(

Can this *very useful* feature be re-added please..? I realise I could use `kernel.json` to specify this sort of thing, but in my case the application allows users to update environment variables on the fly and I want this reflected when they run their code with Jupyter.

Does this make sense..?

**Comment by ntoll** [comment]
FYI... this appears to have happened before:

https://github.com/jupyter/jupyter_client/commit/6406393ff0fcb66fc547f77f81541af1e507978b

:-/

**Comment by takluyver** [comment]
Ah, right. That commit is part of my reworking of the machinery to find and launch kernels, which is not yet released. My current thinking is to turn it into a new model with a different name so that it can be installed alongside jupyter_client and code can gradually transition. But stuff got a bit hectic (I'm moving to Germany tomorrow), so I haven't got round to that yet.

**Comment by ntoll** [comment]
Oooh.... best of luck with the move.

It would be really helpful to know if/when this work will land in the version that's released on PyPI. If it's not going to be for a while, I'll have to bodge another way around this.

*TOTALLY* understand if it takes time to reply.... you have far more important things to deal with at the moment. ;-)

**Comment by takluyver** [comment]
It's probably going to take a while before my ongoing overhaul is ready to use. If we're happy with your fix in #366, that can probably be out quite a bit sooner.

**Comment by ntoll** [comment]
Aha... I've just flown into New York and on the flight over I quickly did https://github.com/mu-editor/mu/commit/a430f41401f1a30956a645a889c2619f32c1a80a which is a workaround. However, if #366 makes it into the released version it will be a preferable solution and I can easily revert the "transatlantic" commit. ;-)

Hope your move is going well..!

**Comment by minrk** [comment]
extra_env was not removed; it has been added and not-yet released in 6.0.
